app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,75 @@
+class DriversController < ApplicationController
+  def index # Fleching out our index action.
+    @drivers = Driver.all
+  end
+
+  def show
+    # Raise an exception.
+    # Be aware that any string will be converted to 0
+    driver_id = params[:id].to_i
+    # It needs to handle this exception.
+    @driver = Driver.find_by(id: driver_id)
+
+    puts @driver
+
+    if @driver.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @driver = Driver.find_by(id: params[:id])
+    if @driver.nil?
+      head :not_found
+      return
+    elsif @driver.update(driver_params)
+      redirect_to driver_path(@driver.id) # go to the index so we can see the driver in the list
+      return
+    else # save failed :(
+      render :edit, status: :bad_request # show the new driver form view again
+      return
+    end
+  end
+
+  def edit
+    @driver = Driver.find_by(id: params[:id])
+
+    if @driver.nil?
+      head :not_found
```

Comment:
It would be better now to do something more sophisticated like redirect the user, instead of a blank screen.